src/art/skypilot/api.py

```diff
@@ -163,9 +186,13 @@ async def _launch_cluster(
         print(task)
 
         try:
-            await to_thread_typed(
-                lambda: sky.launch(task=task, cluster_name=self._cluster_name)
+            job_id, handler = await to_thread_typed(
+                lambda: sky.stream_and_get(
+                    sky.launch(task=task, cluster_name=self._cluster_name)
+                )
             )
+
+            return job_id
```

Comment:
Instead of returning job_id here, would it make sense to start tailing logs within _launch_cluster? That would reduce the amount of code in the main initialize_cluster function, and seems like it should be part of the launching process.